src/connector/src/source/common.rs

```diff
@@ -27,60 +28,81 @@ pub(crate) async fn into_chunk_stream(
     data_stream: impl Stream<Item = ConnectorResult<Vec<SourceMessage>>> + Send + 'static,
     parser_config: ParserConfig,
     source_ctx: SourceContextRef,
+) {
+    // add metrics to the data stream
+    let data_stream = inner_into_chunk_stream(data_stream, source_ctx.clone());
+
+    let parser =
+        crate::parser::ByteStreamSourceParserImpl::create(parser_config, source_ctx).await?;
+    #[for_await]
+    for msg_batch in parser.into_stream(data_stream) {
+        yield msg_batch?;
+    }
+}
+
+#[try_stream(boxed, ok = Vec<SourceMessage>, error = ConnectorError)]
+async fn inner_into_chunk_stream(
+    data_stream: impl Stream<Item = ConnectorResult<Vec<SourceMessage>>> + Send + 'static,
+    source_ctx: SourceContextRef,
 ) {
     let actor_id = source_ctx.actor_id.to_string();
     let fragment_id = source_ctx.fragment_id.to_string();
     let source_id = source_ctx.source_id.to_string();
     let source_name = source_ctx.source_name.to_string();
     let metrics = source_ctx.metrics.clone();
+    let mut partition_input_count = HashMap::new();
+    let mut partition_bytes_count = HashMap::new();
 
-    // add metrics to the data stream
-    let data_stream = data_stream
-        .inspect_ok(move |data_batch| {
```

Comment:
`inspect_ok` takes an immutable closure so that we cannot insert an entry to the hashmap inside it,
so have to create a new `inner_into_chunk_stream` to go through the message once and then yield it back to `parser.into_stream(data_stream)`

Reply:
different from other metrics, the label is determined during runtime when the message is read, which is the root cause of the trouble.

Reply:
> `inspect_ok` takes an immutable closure

Looks like it takes an `FnMut`... 🤔

Reply:
that's weird cannot compile before, anyway I revert it back